rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/WorkspaceInfo.rsc

```diff
@@ -310,6 +315,11 @@ bool rascalIsCollectionType(AType at) = at is arel || at is alrel || at is atupl
 bool rascalIsConstructorType(AType at) = at is acons;
 bool rascalIsDataType(AType at) = at is aadt;
 
+bool rascalIsDataTypeLike(dataId()) = true;
```

Comment:
Oh this is interesting. There are no tests for renaming these 'nodes' at all yet (I was not even aware of their existence). Let me look into this / add some tests.

Reply:
I'm not sure how much the typechecker even support field deref on raw node values.

Reply:
Looked into this for a bit. Currently, there is not support for nodes at all (since we assume that field LHSs refer to ADT or constructor types). We can support this, but I propose to do this in a separate PR, since I foresee quite some changes.
https://github.com/usethesource/rascal-language-servers/issues/510

Reply:
Agreed, better to pick up later.